### ISSUE TITLE
Undefined constant "Gacela\Framework\Config\GLOB_BRACE"

### ISSUE THREAD

**Issue by ilstefano** [gacela-project/gacela-example#4]
Hi,

I get ```Error: Undefined constant "Gacela\Framework\Config\GLOB_BRACE"``` in trying each of the examples.

What I do:

1. clone repo
2. cd example-dir
3. composer install
4. php app.php

As example:

```
stl@stl-symfony /workspace/gacela-example/comment-spam-score (master) $ php app.php 

Fatal error: Uncaught Error: Undefined constant "Gacela\Framework\Config\GLOB_BRACE" in /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/PathFinder.php on line 16

Error: Undefined constant "Gacela\Framework\Config\GLOB_BRACE" in /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/PathFinder.php on line 16

Call Stack:
    0.0004     399216   1. {main}() /workspace/gacela-example/comment-spam-score/app.php:0
    0.0093    1194152   2. Gacela\Framework\Gacela::bootstrap($appRootDir = '/workspace/gacela-example/comment-spam-score', $setup = ???) /workspace/gacela-example/comment-spam-score/app.php:11
    0.0099    1220768   3. Gacela\Framework\Config->init() /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Gacela.php:25
    0.0099    1220768   4. Gacela\Framework\Config->loadAllConfigValues() /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config.php:80
    0.0126    1394448   5. Gacela\Framework\Config\ConfigLoader->loadAll() /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config.php:133
    0.0126    1394800   6. Gacela\Framework\Config\ConfigLoader->readAbsolutePatternPath($pattern = '/workspace/gacela-example/comment-spam-score/config/*.php', $configItem = class Gacela\Framework\Config\GacelaFileConfig\GacelaConfigItem { private string $path = 'config/*.php'; private string $pathLocal = 'config/local.php'; private Gacela\Framework\Config\ConfigReaderInterface $reader = class Gacela\Framework\Config\ConfigReader\PhpConfigReader {  } }, $cacheConfigFileContent = []) /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/ConfigLoader.php:40
    0.0126    1394800   7. Gacela\Framework\Config\PathFinder->matchingPattern($pattern = '/workspace/gacela-example/comment-spam-score/config/*.php') /workspace/gacela-example/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/ConfigLoader.php:89
```

My PHP version is 8.1.5:

```
stl@stl-symfony /workspace/gacela-example/comment-spam-score (master) $ php --version
PHP 8.1.5 (cli) (built: Apr 18 2022 23:52:55) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.5, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.5, Copyright (c), by Zend Technologies
    with Xdebug v3.1.4, Copyright (c) 2002-2022, by Derick Rethans
stl@stl-symfony /workspace/gacela-example/comment-spam-score (master) $ 
```

My OS is Alpine 3.15:

```
stl@stl-symfony /workspace/gacela-example/comment-spam-score (master) $ cat /etc/os-release 
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.15.4
PRETTY_NAME="Alpine Linux v3.15"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"
```

If you want I send you my Dockerfile.

Thanx and cheers,
Stephan

**Comment by Chemaclass** [comment]
Can you try again? I released a new version https://github.com/gacela-project/gacela/releases/tag/0.17.1 with the potential fix. Tell me if you can still reproduce the bug or not anymore.

**Comment by ilstefano** [comment]
No progress, although fresh setup including your changes ... sorry ...

```
stl@stl-symfony /workspace/comment-spam-score (master) $ php app.php 

Fatal error: Uncaught Error: Undefined constant "Gacela\Framework\Config\GLOB_BRACE" in /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/PathFinder.php on line 16

Error: Undefined constant "Gacela\Framework\Config\GLOB_BRACE" in /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/PathFinder.php on line 16

Call Stack:
    0.0005     400824   1. {main}() /workspace/comment-spam-score/app.php:0
    0.0111    1198088   2. Gacela\Framework\Gacela::bootstrap($appRootDir = '/workspace/comment-spam-score', $setup = class Gacela\Framework\Setup\SetupGacela { private $configFn = class Closure { virtual $closure = "{closure}", public $parameter = [...] }; private $mappingInterfacesFn = NULL; private $suffixTypesFn = NULL; private array $externalServices = [] }) /workspace/comment-spam-score/app.php:18
    0.0113    1209608   3. Gacela\Framework\Config->init() /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Gacela.php:25
    0.0113    1209608   4. Gacela\Framework\Config->loadAllConfigValues() /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config.php:80
    0.0147    1377632   5. Gacela\Framework\Config\ConfigLoader->loadAll() /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config.php:133
    0.0147    1377984   6. Gacela\Framework\Config\ConfigLoader->readAbsolutePatternPath($pattern = '/workspace/comment-spam-score/config/*.php', $configItem = class Gacela\Framework\Config\GacelaFileConfig\GacelaConfigItem { private string $path = 'config/*.php'; private string $pathLocal = 'config/local.php'; private Gacela\Framework\Config\ConfigReaderInterface $reader = class Gacela\Framework\Config\ConfigReader\PhpConfigReader {  } }, $cacheConfigFileContent = []) /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/ConfigLoader.php:40
    0.0147    1377984   7. Gacela\Framework\Config\PathFinder->matchingPattern($pattern = '/workspace/comment-spam-score/config/*.php') /workspace/comment-spam-score/vendor/gacela-project/gacela/src/Framework/Config/ConfigLoader.php:89
```

```
stl@stl-symfony /workspace/comment-spam-score (master) $ grep -r 'final class FacaModuleA extends AbstractFacade' *
vendor/gacela-project/gacela/tests/Feature/Framework/UsingCustomSuffixTypes/ModuleA/FacaModuleA.php:final class FacaModuleA extends AbstractFacade
```

**Comment by Chemaclass** [comment]
Can you please share your Dockerfile? I am using MacOS 12.3.1 - PHP 8.1.4 (cli), and I cannot reproduce the error. 

**UPDATE**:
Ah, I found the root of the problem. See: https://www.php.net/manual/en/function.glob.php
```bash
Note: The GLOB_BRACE flag is not available on some non GNU systems, like Solaris or Alpine Linux.
```
> Other related issue: https://github.com/UnionOfRAD/lithium/issues/1291

Maybe with your Dockerfile I can reproduce the same bug and find a solution for it :)

**Comment by Chemaclass** [comment]
I can reproduce it with this Dockerfile:
```Dockerfile
FROM php:8.1-fpm-alpine
WORKDIR /usr/src/app

# install composer
RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
RUN php composer-setup.php --install-dir=/usr/local/bin --filename=composer
RUN php -r "unlink('composer-setup.php');"

# copy files
COPY ./ /usr/src/app/
 
# install packages
ENV COMPOSER_ALLOW_SUPERUSER 1
RUN composer install
```

And then:
```bash
docker build -t comment-spam-score .
docker run -ti comment-spam-score sh
php app.php

comment-spam-score/ (master) $ docker run -ti comment-spam-score sh
/usr/src/app # php app.php 

Fatal error: Uncaught Error: Undefined constant "Gacela\Framework\Config\GLOB_BRACE" in /usr/src/app/vendor/gacela-project/gacela/src/Framework/Config/PathFinder.php:16
```

I will work on the fix now 🚀

**Comment by Chemaclass** [comment]
We created a new released, @ilstefano. Tell us if you can still reproduce the error or you discover something new 🥳
https://github.com/gacela-project/gacela/releases/tag/0.17.2

**Comment by JesusValeraDev** [comment]
You can `git pull` from `master`, it should work now :)

**Comment by ilstefano** [comment]
Hi, 
yes, 0.17.2 fixed it, great, thanx a lot!

**Comment by Chemaclass** [comment]
@ilstefano I created a Dockerfile with linux alpine https://github.com/gacela-project/gacela/pull/147 because it's the smaller linux container I found, and it could be nice to run gacela tests or develop anything for gacela without having PHP in your local machine. What do you think?

**Comment by ilstefano** [comment]
Hi, I did a short 'code review' within the PR. Hope this is how you intended to do so ...
Basically I think this is the right way for cloud native development! You next can easily extend your Dockerfile to a rough build and deployment tool.

Besides a docker image I'm also using VSC as IDE and its remote container features, especially devcontainers. This is extremely productive ... you can use lots of configurations, runtimes, libraries, tools and platform services (like database, ...) isolated from each other, gitops-like version controlled and runnable just with one click: 'Reopen in container'!

**Comment by ilstefano** [comment]
Excuse me, I just realize that I have much more 2cts on this topic.

I would at least also provide XDebug for debugging in the image. And there should be some user configuration in the development image such that the container process has same permissions as the host user. If you like I send you my typical Dockerfiles for PHP development and some links what other people's best practices are.

**Comment by Chemaclass** [comment]
@ilstefano that would be awesome. You can share it as a comment or you can create a PR with your suggestions, so you'll be a collaborator of the project too :)

I really don't mind closing this PR if you create another one. After all, this was your idea somehow.

**Comment by ilstefano** [comment]
I'm looking forward to create a WiP-PR next days ... cu

**Comment by ilstefano** [comment]
Hi @Chemaclass,

sorry for the delay ... but it took me a bit time to organize my head about how and where to integrate docker stuff into gacela. And I also had to visit the Spryker's Bootcamp  in Berlin yesterday :-)

So I finally came up with the idea that it makes more sense for your users to integrate docker means into this repo, the example repo, and not foremost into the framework repo itself.

As I'm not sure whether you join my thoughts and I do not want pollute your repo with unwanted PRs I suggest that you have first a look into my fork: https://github.com/ilstefano/gacela-example/pull/1

If this meets your concepts than I will immediately rise a PR :-) !

What I've also already prepared ... but what takes still a bit a cleaner implementation as an easy to use and clear extension of standard docker is the usage in Visual Studio Code, which is really amazing as I already wrote.

So I will put VSC in another branch and then ping you again ...

Cheers,
Stephan
